core/forkid/forkid.go

```diff
@@ -203,6 +201,28 @@ func newFilter(config *params.ChainConfig, genesis common.Hash, headfn func() ui
 			// No exact, subset or superset match. We are on differing chains, reject.
 			return ErrLocalIncompatibleOrStale
 		}
+
+		head, time := headfn()
+		// Verify forks by block
+		for i, fork := range forks {
+			// If our head is beyond this fork, continue to the next (we have a dummy
+			// fork of maxuint64 as the last item to always fail this check eventually).
+			if head >= fork {
+				continue
+			}
+			return verify(i, head)
+		}
+		// Verify forks by time
+		for i := len(forks); i < len(forks)+len(forksByTime); i++ {
+			fork := forksByTime[i-len(forks)]
```

Comment:
Looks odd, Shouldn't it be
```suggestion
			fork := allForks[i-len(forks)]
```
?

Reply:
yes

Reply:
Hmm no, the `i` goes from `len(forks)...len(forks)+len(forksByTime)`, so in order to correctly index `forksByTime` we need to substract `len(forks)`. Could be switched to `allForks[i]` but not `allForks[i-len(forks)]`

Reply:
Ah, you are right. For legibility, I think you should change it to be
```
for i := 0; i < len(forksByTime), i++{
    fork := forksByTime[i]
   ...
    verify(i + len(forks), time) 
...

Then it can be further simplified
```
for i, fork := range forksByTime { 
... 
    verify(i + len(forks), time) 
```